packages/components/copybutton/src/CopyButton.tsx

```diff
@@ -77,21 +83,33 @@ function _CopyButton(props: CopyButtonProps, ref: React.Ref<HTMLDivElement>) {
       className={cx(styles.wrapper, className)}
       {...otherProps}
     >
-      <CopyToClipboard text={value} onCopy={handleOnCopy}>
-        <Tooltip
-          content={copied ? tooltipCopiedText : tooltipText}
-          {...tooltipProps}
+      {isDisabled ? (
+        <button
+          type="button"
+          ref={button}
+          className={cx(styles.copyButton, styles.copyButtonDisbaled)}
+          aria-label={`Copy ${value} to clipboard`}
+          disabled={isDisabled}
         >
-          <button
-            type="button"
-            ref={button}
-            className={cx(styles.copyButton)}
-            aria-label={`Copy ${value} to clipboard`}
+          <CopyIcon variant="muted" />
+        </button>
+      ) : (
+        <CopyToClipboard text={value} onCopy={handleOnCopy}>
+          <Tooltip
+            content={copied ? tooltipCopiedText : tooltipText}
+            {...tooltipProps}
           >
-            <CopyIcon variant="muted" />
-          </button>
-        </Tooltip>
-      </CopyToClipboard>
+            <button
+              type="button"
+              ref={button}
+              className={cx(styles.copyButton)}
+              aria-label={`Copy ${value} to clipboard`}
+            >
+              <CopyIcon variant="muted" />
+            </button>
+          </Tooltip>
+        </CopyToClipboard>
+      )}
```

Comment:
Instead of doing all this stuff

I think you can pass `disabled={isDisabled}` to the button and to the tooltip you pass `content=""` and it won't render it

Yes I know, this is not great DX
we should change this once we migrate the tooltip to v4
Or have we already done that?
If yes, we can just make it possible to force the tooltip not render

Reply:
Ideally, we should introduce `isDisabled` to a tooltip component and it it's true, we should just render `children`

Reply:
yeah, good idea. Thanks!